overlord/snapstate/handlers.go

```diff
@@ -345,6 +351,7 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 	if conflErr, ok := err.(*ChangeConflictError); ok {
 		// conflicted with an install in the same change, just skip
 		if conflErr.ChangeID == t.Change().ID() {
```

Comment:
[This was added 3 years ago](https://github.com/snapcore/snapd/blame/master/overlord/snapstate/handlers.go#L345-L349) already as part of [this PR](https://github.com/snapcore/snapd/pull/10849)

This change works for the specific case [described here](https://github.com/snapcore/snapd/pull/10849#issue-1008392308), but is arguable dangerous in terms of the ability to causing silent failure.

- One improvement would be to at least issue warning when this happens so that the reason for downstream failure is obvious to the user.
- Could the conflict check perhaps take into account if the tasks associated with a conflicting snap was completed (e.g.  after retries). From the description where A continues to install dependencies B, but still the conflict remains.. this superficially seems a little odd.